CMakeLists.txt

```diff
@@ -458,8 +458,15 @@ endif()
 
 # Macro to check if source compiles when cross-compiling
 # or runs when compiling natively
+#
+# Note: because we want to match the behavior
```

Comment:
I'm not sure if this CMake comment is really necessary. It seems like this change actually fixes a bug in non-native compiles.

Reply:
In your comment you mention that configure doesn't make runtime checks like this. But I looked at the try() function and it does check to see that the output is zero. Did you mean something else? Sorry I am trying to understand.

Reply:
aye, try() verifies that the given command runs... but the given command is a compile command; it does not also run the output of the compile command.

Reply:
If that is the case then maybe we should only be using `check_c_source_compiles` in CMake. What do you think?

Reply:
I think it's reasonable to let the native flag drive slightly divergent behavior for now, since it's already a "here's a flag to shoot yourself in the foot" kind of thing.

Reply:
(Users who turn on the native flag want the very tightest, and who are we to deny them?
But if this lands, it would be reasonable to file a bug against configure, saying it's missing
a chance to omit dead code in the --native case.)